apps/web/src/contexts/globalLayout/GlobalSidebar/StandardSidebar.tsx

```diff
@@ -170,13 +173,22 @@ export function StandardSidebar({ queryRef }: Props) {
       content: <PostComposerModalWithSelector viewerRef={query?.viewer} queryRef={query} />,
       headerVariant: 'thicc',
       isFullPage: isMobile,
-      onCloseOverride: () => {
+      onCloseOverride: (onClose: () => void) => {
+        if (!captionRef.current) {
+          onClose();
+          return;
+        }
```

Comment:
cool change here: in the past, `onCloseOverride` simply stopped the modal from closing then left it up to the dev to figure out which modal to close. this required us to do something like:

```js
onCloseOverride: () => {
  // ...some logic, and then
  hideModal({ id: 'post-composer' })
}
```

which required mental overhead of remembering which modal to close. most of the time, you simply want to close the modal that's open – but calling `hideModal()` without an ID may close the wrong modal, if multiple are open.

the improvement was to pass in the relevant hideModal callback as an argument to the function, allowing you to reliably close the modal when you'r ready:

```js
onCloseOverride: (onClose) => {
  // ...some logic, and then
  onClose()
}
```

Reply:
🧠🧠